src/hypothesis/searchstrategy/types.py

```diff
@@ -0,0 +1,357 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import re
+import functools
+import collections
+from inspect import isclass
+
+import hypothesis.strategies as st
+from hypothesis.errors import InvalidArgument
+from hypothesis.internal.compat import text_type, integer_types, \
+    getfullargspec
+
+_global_type_to_strategy_lookup = {}
+
+
+def get_type_hints(thing):
+    """Try to return any type hints for ``thing``."""
+    try:
+        import typing
+        return typing.get_type_hints(thing)
+    except TypeError:
+        # `thing` is not a module, class, method, or function
+        return {}
+    except (ImportError, AttributeError):  # pragma: no cover
+        # This is a fallback for Python <3.6, where get_type_hints may fail
+        try:
+            spec = getfullargspec(thing)
+            return {
+                k: v for k, v in spec.annotations.items()
+                if k in (spec.args + spec.kwonlyargs) and isinstance(v, type)
+            }
+        except TypeError:
+            return {}
+
+
+def required_args(spec):
+    """Return a set of required args in this argspec."""
+    return set(spec.args + spec.kwonlyargs) \
+        - set(spec.args[len(spec.args) - len(spec.defaults or ()):]) \
+        - set(spec.kwonlydefaults or ())
+
+
+def from_type(thing):
+    """Resolve a type to an appropriate search strategy.
+
+    1. If ``thing`` is a user-registered type or in the builtin lookup,
+       return the corresponding strategy.
+
+    2. If ``thing`` is a subclass of something from the typing module, the
+       corresponding strategy is returned.  For abstract types this may be
+       the union of one or more strategies for concrete subtypes.
+
+    3. If ``thing`` has a subtype in the lookup mapping you
```

Comment:
There's no longer a supplied lookup function to this argument, so this entry isn't relevant I think.